src/applications/renderer/src/components/legend/component.js

```diff
@@ -45,12 +47,17 @@ const Legend = ({
         alias: option.label !== option.value ? option.label : undefined,
       };
 
-    if (isPie) {
-      patchConfiguration({ category: newOption });
+    if (isPie || isDonut) {
+      patchConfiguration({
+        category: newOption,
+        color: newOption
```

Comment:
This line can be deleted.

If you create a bar chart with, let's say, “Country” on the X axis (category) and “Gross GDP” on the Y (value), when you switch to the pie/donut view, it is rendered correctly.

But if you start with the pie/donut chart instead, and select “Gross GDP” below the chart (value) and “Country” in the legend (category), when you switch to the bar chart, the colour field also gets set. This is unneeded and breaks the symmetry of the chart building process.